File: Leetcode/Daily/06-2025/Java/19.java
// Problem: 2294. Partition Array Such That Maximum Difference Is K (Medium)
// Link: https://leetcode.com/problems/partition-array-such-that-maximum-difference-is-k

class Solution {
    public int partitionArray(int[] nums, int k) {
        int ans = 1, n = nums.length;
        Arrays.sort(nums);
        int curr = nums[0];

        for (int i = 1; i < n; i++) {
            if (nums[i] - curr > k)  {
                ans++;
                curr = nums[i];
            }
        }

        return ans;        
    }
}

File: Leetcode/Dailies/2025/02/19.cpp
// Problem: 1415. The k-th Lexicographical String of All Happy Strings of Length n (Medium)
// Link: https://leetcode.com/problems/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n

class Solution {
public:
    string getHappyString(int n, int k) {
        k--;
        int i = 1 << (n-1);
        
        if (k >= 3 * i) return "";

        string ans;

        while (ans.size() < n) {
            int num = k / i;
            k %= i;
            i /= 2;

            if (ans.empty()) ans.push_back('a' + num);
            else {
                if (num == 0) {
                    if (ans.back() == 'a') ans.push_back('b');
                    else ans.push_back('a');
                } if (num == 1) {
                    if (ans.back() == 'c') ans.push_back('b');
                    else ans.push_back('c');
                }
            }
        }

        return ans;
    }
};